static/index.js

```diff
@@ -436,8 +454,16 @@ $("#btnroomusers").on('click', function() {
 
 socket.on('chatm', function(d){
     d=JSON.parse(d);
+    // If scrolled to bottom already, scroll to bottom again after appending message
+    var atBottom = false;
+    if($(window).scrollTop() + $(window).height() == $(document).height()) {
+      atBottom = true;
+    }
     appendChatMessage(d.uid,d.room,d.name,d.nick,d.m,d.msgid,d.avatar,d.time);
-    scrollToBottom();
+    if (atBottom) {
+      scrollToBottom();
+    }
+
```

Comment:
I think for this to be done 'clean', scrollToBottom() should be added to appendChatMessage and appendSystemMessage, and that logic could be added in there as well. Also, have you tested this still works when using a mobile device and the keyboard is pulled up, etc?